### PR TITLE
Adds control plane service object  for cluster endpoint

### PR DESCRIPTION
**Pull request by prateekgogia**
*Issue #, if available:*

*Description of changes:*
Create a Kubernetes service object of type Loadbalancer for cluster endpoint 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: operator/pkg/controllers/master/master.go [hunk 1]
```diff
@@ -69,28 +77,76 @@ func (c *Controller) createSecrets(ctx context.Context, controlPlane *v1alpha1.C
 	return nil
 }
 
-func certListFor(controlPlane *v1alpha1.ControlPlane) []*secrets.Request {
+func (c *Controller) getControlPlaneHostname(ctx context.Context, controlPlane *v1alpha1.ControlPlane) (string, error) {
```
Comment (ellistarn) [inline operator/pkg/controllers/master/master.go:80]: Seems odd to me that the "get" method would "create" internally. 

I might model this like:
```
service, err := c.client.ensure(ctx, c.ServiceFor(controlPlane))
hostname := service.Status.Loadbalancer.Ingress
if  == nil {
  return result.Wait()
}

c.SecretsFor(hostname)
...
```

File: operator/pkg/controllers/master/master.go [hunk 2]
```diff
@@ -69,28 +77,76 @@ func (c *Controller) createSecrets(ctx context.Context, controlPlane *v1alpha1.C
...
+	hostname, err := c.createService(ctx, controlPlane)
+	if err != nil {
+		return "", err
+	}
+	if hostname == "" {
+		return "", fmt.Errorf("waiting for control plane hostname, %w", errors.WaitingForSubResources)
```
Comment (ellistarn) [inline operator/pkg/controllers/master/master.go:86]: In the long run, let's model this as a reconcile.result instead of munging it into a typed error.

**Review by ellistarn — APPROVED** [review]

File: operator/pkg/controllers/controller.go [hunk 3]
```diff
@@ -21,7 +21,7 @@ import (
 
 	"github.com/awslabs/kit/operator/pkg/apis/infrastructure/v1alpha1"
 	"github.com/awslabs/kit/operator/pkg/errors"
-	"github.com/awslabs/kit/operator/pkg/status"
+	"github.com/awslabs/kit/operator/pkg/result"
```
Comment (JacobGabrielson) [inline operator/pkg/controllers/controller.go:24]: I'd suggest renaming this pkg to `results` maybe? It's that that `result` is such a common word to actually use (especially in Go), even if you don't have actual naming conflicts, every time someone sees `result` they're going to thing it's a local variable (containing a result).

File: operator/pkg/controllers/controller.go [hunk 4]
```diff
@@ -48,40 +48,43 @@ func (c *GenericController) Reconcile(ctx context.Context, req reconcile.Request
...
+		}
+		return *result.Failed, reconcileErr
 	}
-	return result, nil
+	return reconcileResult, nil
 }
 
 func (c *GenericController) reconcile(ctx context.Context, resource Object, persisted runtime.Object) (reconcile.Result, error) {
-	var result *reconcile.Result
+	var reconcileResult *reconcile.Result
```
Comment (JacobGabrielson) [inline operator/pkg/controllers/controller.go:71]: See earlier comment - I think having to do this renaming is probably a sign that that `result` pkg's name should change instead.

File: operator/pkg/controllers/controller.go (hunk 3 above)
Comment (prateekgogia) [reply operator/pkg/controllers/controller.go:24]: Thanks, yeah I will address this as part of the next PR I am working on.

**Pull request merged**